### ISSUE TITLE
Could not load libjvm on Some ARM-based Windows 11 24H2

### ISSUE THREAD

**Issue by kdaek21** [ikvmnet/ikvm#627]
I have two Arm-based Windows devices.
One is Galaxy Book2 (SM-W737 model) and the other is Galaxy Book Edge (NT960XMB-KC01B model) that was recently released.

Galaxy Book2 is an old device that was released recently, so it was Windows 11 23H2 until recently, and recently upgraded to Windows 11 24H2. Galaxy Book Edge was released as Copilot+PC from the time of release, although it was updated recently, so it was released as Windows 11 24H2.

I haven't checked the previous version, but ikvm-8.10.3-net8.0-arm64 version does not work on Galaxy Book2 that was updated to Windows 11 24H2, causing 'Could not load libjvm' error. (It works normally on Galaxy Book Edge, and I couldn't confirm whether it worked normally before updating to Windows 11 24H2.)

The specifications for each are as follows. 
CPU: Galaxy Book 2 - Qualcomm Snapdragon 850 / Galaxy Book Edge - Qualcomm Snapdragon X Elite
RAM: Galaxy Book 2 - 4GB / Galaxy Book Edge - 16GB

The error message is as follows, but the same error was registered in a different case #623. Is it the same problem? Or is it a compatibility issue between the CPU (or the entire device) and the updated Windows (will it be resolved with a future Windows update?)?

Unhandled exception. System.TypeInitializationException: The type initializer for 'java.lang.Throwable' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'IKVM.Runtime.LibJava' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'IKVM.Runtime.LibJvm' threw an exception.
 ---> IKVM.Runtime.InternalException: Could not load libjvm.
   at IKVM.Runtime.LibJvm..ctor() in /_/src/IKVM.Runtime/LibJVM.cs:line 137
   at IKVM.Runtime.LibJvm..cctor() in /_/src/IKVM.Runtime/LibJVM.cs:line 105
   --- End of inner exception stack trace ---
   at IKVM.Runtime.LibJava..ctor() in /_/src/IKVM.Runtime/LibJava.cs:line 30
   at IKVM.Runtime.LibJava..cctor() in /_/src/IKVM.Runtime/LibJava.cs:line 22
   --- End of inner exception stack trace ---
   at IKVM.Runtime.BootstrapClassLoader..ctor(RuntimeContext context) in /_/src/IKVM.Runtime/BootstrapClassLoader.cs:line 49
   at IKVM.Runtime.RuntimeClassLoaderFactory.GetBootstrapClassLoader() in /_/src/IKVM.Runtime/RuntimeClassLoaderFactory.cs:line 134
   at IKVM.Runtime.RuntimeAssemblyClassLoaderFactory.Create(Assembly assembly) in /_/src/IKVM.Runtime/RuntimeAssemblyClassLoaderFactory.cs:line 123
   at IKVM.Runtime.RuntimeAssemblyClassLoaderFactory.FromAssembly(Assembly assembly) in /_/src/IKVM.Runtime/RuntimeAssemblyClassLoaderFactory.cs:line 81
   at IKVM.Runtime.RuntimeClassLoaderFactory.GetJavaTypeFromType(Type type) in /_/src/IKVM.Runtime/RuntimeClassLoaderFactory.cs:line 250
   at IKVM.Java.Externs.ikvm.runtime.Util.getFriendlyClassFromType(Type type) in /_/src/IKVM.Runtime/Java/Externs/ikvm/runtime/Util.cs:line 125
   at ikvm.runtime.Util.getFriendlyClassFromType(Type type)
   at java.lang.Class.op_Implicit(Type type) in Class.java:line 175
   at IKVM.Runtime.ExceptionHelper.GetPersistentFields() in /_/src/IKVM.Runtime/ExceptionHelper.cs:line 430
   at java.lang.Throwable..cctor()
   --- End of inner exception stack trace ---
   at java.lang.Throwable.__mapImpl(Exception)
   at IKVM.Runtime.ExceptionHelper.MapException[T](Exception e, Boolean remap, Boolean unused) in /_/src/IKVM.Runtime/ExceptionHelper.cs:line 925
   at IKVM.Runtime.ExceptionHelper.MapTypeInitializeException(TypeInitializationException t, Type handler) in /_/src/IKVM.Runtime/ExceptionHelper.cs:line 865
   at IKVM.Runtime.ExceptionHelper.MapException[T](Exception e, Boolean remap, Boolean unused) in /_/src/IKVM.Runtime/ExceptionHelper.cs:line 919
   at IKVM.Runtime.ExceptionHelper.MapTypeInitializeException(TypeInitializationException t, Type handler) in /_/src/IKVM.Runtime/ExceptionHelper.cs:line 865
   at IKVM.Runtime.ExceptionHelper.MapException[T](Exception e, Boolean remap, Boolean unused) in /_/src/IKVM.Runtime/ExceptionHelper.cs:line 919
   at IKVM.Java.Externs.ikvm.runtime.Util.mapException(Exception e) in /_/src/IKVM.Runtime/Java/Externs/ikvm/runtime/Util.cs:line 194
   at ikvm.runtime.Util.mapException(Exception x)
   at IKVM.Runtime.Launcher.Run(Assembly assembly, String main, Boolean jar, String[] args, String rarg, IDictionary`2 properties) in /_/src/IKVM.Runtime/Launcher.cs:line 556
   at IKVM.Tools.Java.Program.Main(String[] args) in /_/src/java/Program.cs:line 11

**Comment by wasabii** [comment]
Might be the same. If it's missing the required libraries, it's missing the required libraries.

But other than that, you'll have to tel me why the DLLs can't load. Look at them in depwalker or something?

**Comment by kdaek21** [comment]
I just checked and it was the same problem.
I downloaded and installed the latest VC++ redistributable package for arm64 from the official MS website and confirmed normal operation.

When I checked the Galaxy Book Edge, there was a trace that I had installed the latest VC++ redistributable package.

It seems that the previously installed version was changed to the default installation version of Win 24H2, not the latest, when I updated to Win 24H2 (I saw the vcruntime140.dll file in the System32 folder).
